packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -103,11 +97,31 @@ + (BOOL)requiresMainQueueSetup {
     return @[ @"appearanceChanged" ];
 }
 
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
```

Comment:
is it worth checking if we are already on mainthread just return the current thing and if we are on the background thread use get the main queue kind of logic? or are you trying to always do it on the next main queue cycle?

Reply:
This bit of code comes from https://reactnative.dev/docs/native-modules-ios#threading 

Documentation on methodQueue:

```objective-c
/**
 * The queue that will be used to call all exported methods. If omitted, this
 * will call on a default background queue, which is avoids blocking the main
 * thread.
 *
 * If the methods in your module need to interact with UIKit methods, they will
 * probably need to call those on the main thread, as most of UIKit is main-
 * thread-only. You can tell React Native to call your module methods on the
 * main thread by returning a reference to the main queue, like this:
 *
 * - (dispatch_queue_t)methodQueue
 * {
 *   return dispatch_get_main_queue();
 * }
 *
 * If you don't want to specify the queue yourself, but you need to use it
 * inside your class (e.g. if you have internal methods that need to dispatch
 * onto that queue), you can just add `@synthesize methodQueue = _methodQueue;`
 * and the bridge will populate the methodQueue property for you automatically
 * when it initializes the module.
 */
@property (nonatomic, strong, readonly) dispatch_queue_t methodQueue;
```

Without this code (if I comment out this block) I get this error

<img width="1022" alt="Screenshot 2023-02-16 at 1 47 27 PM" src="https://user-images.githubusercontent.com/78454019/219494483-4dc543b7-046c-4abb-9364-ce2bafd49e65.png">

"is it worth checking if we are already on mainthread just return the current thing and if we are on the background thread use get the main queue kind of logic" do you mean something like this?

<img width="1003" alt="Screenshot 2023-02-16 at 1 45 46 PM" src="https://user-images.githubusercontent.com/78454019/219494579-7bd07f6f-6d6e-4c1a-9149-f7b9dd1be8f9.png">

Reply:
I always thought this method was a "What thread should I run native code on?". So I'm not sure that check makes sense. This is more of a declaration "Run this module on X thread".

Reply:
Yeah, this is a necessary bit of native module logic and should not be changed.